core/templates/services/validators.service.spec.ts

```diff
@@ -24,26 +24,20 @@ import { ValidatorsService } from 'services/validators.service';
 
 describe('Validators service', () => {
   let vs: ValidatorsService = null;
-  /* eslint-disable dot-notation */
   const INVALID_NAME_CHARS_COPY = (
-    Array.from(AppConstants['INVALID_NAME_CHARS']));
-  /* eslint-enable dot-notation */
+    Array.from(AppConstants.INVALID_NAME_CHARS));
 
   beforeEach(() => {
     TestBed.configureTestingModule({
       providers: [AlertsService, NormalizeWhitespacePipe]
     });
     vs = TestBed.get(ValidatorsService);
-    /* eslint-disable dot-notation */
-    AppConstants['INVALID_NAME_CHARS'] = '#xyz';
-    /* eslint-enable dot-notation */
+    AppConstants.INVALID_NAME_CHARS = ['#', 'x', 'y', 'z'];
```

Comment:
Why is this possible when `AppConstants` is `const`?

Reply:
Refer -> https://stackoverflow.com/a/23436563/13135203

Reply:
Do you think we should be using Object.freeze() (as mentioned on that page) for our list/dict constants in general?

Reply:
I think that it should be possible. Although I can imagine that it can mess up with TypeScript.

Reply:
I had an alternative solution i.e. adding `as const` to constants. Here's what will be the difference.

### Current
```typescript
export const AppConstants = {
  SOME_CONSTANT: ['value']
};
```

Here the type of AppConstants.SOME_CONSTANT is `string[]`.

### Using as const
```typescript
export const AppConstants = {
  SOME_CONSTANT: ['value']
} as const;
```
Here the type of AppConstants.SOME_CONSTANT is `readonly ['value']`.

Doing this has two benefits 
- Readonly. The value of the constant can't be changed now.
- Literal Type. That i.e. the type is not `string[]` but `['value']`. Which makes sense because it is a constant.

Reply:
/cc @vojtechjelinek @srijanreddy98 @seanlip wdyt?

Reply:
SG!